File: fixtures/typescript/src/cat.ts
/**
 * 猫类 - 继承自动物
 */
import { Animal } from './animal';

export class Cat extends Animal {
    private color: string;
    
    constructor(name: string, age: number, color: string) {
        super(name, age);
        this.color = color;
    }
    
    makeSound(): void {
        console.log(`${this.name} says: Meow! Meow!`);
    }
    
    climb(): void {
        console.log(`${this.name} is climbing a tree`);
    }
    
    getColor(): string {
        return this.color;
    }
    
    toString(): string {
        return `Cat(name=${this.name}, age=${this.age}, color=${this.color})`;
    }
}

File: fixtures/python/cat.py
"""
猫类 - 继承自动物
"""
from animal import Animal


class Cat(Animal):
    """猫类"""
    
    def __init__(self, name: str, age: int, color: str):
        super().__init__(name, age)
        self.color = color
    
    def make_sound(self) -> None:
        """猫叫"""
        print(f"{self.name} says: Meow! Meow!")
    
    def climb(self) -> None:
        """爬树"""
        print(f"{self.name} is climbing a tree")
    
    def get_color(self) -> str:
        return self.color
    
    def __str__(self) -> str:
        return f"Cat(name={self.name}, age={self.age}, color={self.color})"
